src/libraries/System.Collections/tests/Generic/Queue/Queue.Generic.Tests.cs

```diff
@@ -289,6 +289,53 @@ public void Queue_Generic_TryDequeue_AllElements(int count)
             }
         }
 
+        [Fact]
+        public void Queue_Generic_IterateLastIndexOfMaxSizedQueue_DoesNotOverflow()
+        {
+            Queue<byte> queue;
+            try
+            {
+                queue = new Queue<byte>(Array.MaxLength);
+            }
+            catch (OutOfMemoryException)
+            {
+                // just skip when ctor throws OOM
+                return;
```

Comment:
```suggestion
                throw new SkipTestException("Unable to allocate 2GB of memory");
```

Reply:
I tried that, but `SkipException` still fails as test runner exits with non-zero code. (it was apparently dropped in commit https://github.com/xunit/xunit/commit/2deeff5cb60bdb8c12fd4f3cfaa38035cdbc7ba6)

Reply:
Even with the ConditionalFact?

Reply:
Ah, I was using `SkipException` from `Xunit.Sdk` (which apparently provides the type definition but doesn't skip). I will test with `SkipTestException` which should work as we have other usages. :)

Reply:
It worked. Thanks!